spec/unit/outputs/elasticsearch_spec.rb

```diff
@@ -78,6 +80,55 @@
           expect(manticore_host).to include(options["path"])
         end
       end
+
+      context "with a URI based path" do
+        let(:options) do
+          o = super()
+          o.delete("path")
+          o["hosts"] = ["http://localhost:9200/mypath/"]
+          o
+        end
+        let(:client_host_path) { URI.parse(eso.client.client_options[:hosts].first).path }
+
+        it "should initialize without error" do
+          expect { eso }.not_to raise_error
+        end
+
+        it "should use the URI path" do
+          expect(client_host_path).to eql("/mypath/")
+        end
+
+        context "with a path option but no URL path" do
+          let(:options) do
+            o = super()
+            o["path"] = "/override/"
+            o["hosts"] = ["http://localhost:9200"]
+            o
+          end
+
+          it "should initialize without error" do
+            expect { eso }.not_to raise_error
+          end
+
+          it "should use the option path" do
+            expect(client_host_path).to eql("/override/")
+          end
+        end
+
+        context "with a path option and a URL path" do
+          let(:do_register) { false } # Register will fail
+          let(:options) do
+            o = super()
+            o["path"] = "/override"
+            o["hosts"] = ["http://localhost:9200/mypath/"]
+            o
+          end
+
+          it "should initialize with an error" do
+            expect { eso.register }.to raise_error(LogStash::ConfigurationError)
+          end
```

Comment:
It raises a ConfigurationError saying that you can't have a path in the URL and in the override.

Would would the other example check? `#register` does not complete in this case.